SQRLDotNetClientUI/Platform/OSX/NotifyIcon.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Avalonia.Controls;
+using SQRLDotNetClientUI.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using MonoMac.AppKit;
+using Avalonia;
+using Avalonia.Platform;
+using Avalonia.Threading;
+
+namespace SQRLDotNetClientUI.Platform.OSX
+{
+    /// <summary>
+    /// Represents a Notification Tray Icon (on a OSxMac StatusBarItem) 
+    /// </summary>
+    public class NotifyIcon : INotifyIcon
+    {
+        private NSStatusItem _item;
+        private NSStatusItem statusBarItem
+        {
+            get => _item; set
+            {
+                _item = value;
+                UpdateMenu();
+            }
+        }
```

Comment:
Should this be public?

Reply:
Hmmm not sure what we would want with it? I can make it public easily enough but it on'y pertains to the Mac

Reply:
Nevermind, Jose. I've just never seen getters and setters on private fields before, so I thought this might have been by mistake. So just forget that I've mentioned it. 😉

Reply:
Reason for the getter setter was to trigger the UpdateMenu() maybe overkill...

Reply:
Not at all, it was just my ignorance. I didn't even know that setters and getters were a thing for private fields. So please nevermind!